src/Components/Blazor/WebAssembly.Authentication/src/Services/RemoteAuthenticationService.cs

```diff
@@ -0,0 +1,188 @@
+            RemoteAuthenticationContext<TRemoteAuthenticationState> context)
+        {
+            await EnsureAuthService();
+            var result = await _jsRuntime.InvokeAsync<RemoteAuthenticationResult<TRemoteAuthenticationState>>("AuthenticationService.signIn", context.State);
+            if (result.Status == RemoteAuthenticationStatus.Success)
+            {
+                UpdateUser(GetCurrentUser());
+            }
+
+            return result;
+        }
+
+        /// <inheritdoc />
+        public virtual async Task<RemoteAuthenticationResult<TRemoteAuthenticationState>> CompleteSignInAsync(
+            RemoteAuthenticationContext<TRemoteAuthenticationState> context)
+        {
+            await EnsureAuthService();
+            var result = await _jsRuntime.InvokeAsync<RemoteAuthenticationResult<TRemoteAuthenticationState>>("AuthenticationService.completeSignIn", context.Url);
+            if (result.Status == RemoteAuthenticationStatus.Success)
+            {
+                UpdateUser(GetCurrentUser());
+            }
+
+            return result;
+        }
+
+        /// <inheritdoc />
+        public virtual async Task<RemoteAuthenticationResult<TRemoteAuthenticationState>> SignOutAsync(
+            RemoteAuthenticationContext<TRemoteAuthenticationState> context)
+        {
+            await EnsureAuthService();
+            var result = await _jsRuntime.InvokeAsync<RemoteAuthenticationResult<TRemoteAuthenticationState>>("AuthenticationService.signOut", context.State);
+            if (result.Status == RemoteAuthenticationStatus.Success)
+            {
+                UpdateUser(GetCurrentUser());
+            }
+
+            return result;
+        }
+
+        /// <inheritdoc />
+        public virtual async Task<RemoteAuthenticationResult<TRemoteAuthenticationState>> CompleteSignOutAsync(
+            RemoteAuthenticationContext<TRemoteAuthenticationState> context)
+        {
+            await EnsureAuthService();
+            var result = await _jsRuntime.InvokeAsync<RemoteAuthenticationResult<TRemoteAuthenticationState>>("AuthenticationService.completeSignOut", context.Url);
+            if (result.Status == RemoteAuthenticationStatus.Success)
+            {
+                UpdateUser(GetCurrentUser());
+            }
+
+            return result;
+        }
+
+        /// <inheritdoc />
+        public virtual async ValueTask<AccessTokenResult> GetAccessToken()
+        {
+            await EnsureAuthService();
+            return await _jsRuntime.InvokeAsync<AccessTokenResult>("AuthenticationService.getAccessToken");
+        }
+
+        /// <inheritdoc />
+        public virtual async ValueTask<AccessTokenResult> GetAccessToken(AccessTokenRequestOptions options)
+        {
+            await EnsureAuthService();
+            return await _jsRuntime.InvokeAsync<AccessTokenResult>("AuthenticationService.getAccessToken", options);
```

Comment:
In the event where the user didn't provide a return URL, what do you think about defaulting to using the current URL as return URL? That's nearly always what the developer would want, isn't it?

What's the current end-to-end behavior if there's no return URL? Where to they come back to?

From reading the code I haven't managed to make sense of the end-to-end flow. It doesn't look like the return URL from  `AccessTokenRequestOptions` actually gets used on the TypeScript side. Does it get used anywhere else?

Reply:
done